### DISCUSSION TITLE
sql: expected 2 arguments, got 1

### DISCUSSION THREAD

**Discussion by Mnemonics** [godror/godror#227] [Q&A]
We use an SQL parameter multiple times in a Query (Oracle).
For instance:

`	SELECT column FROM table
	WHERE startdate < TO_DATE(:1, 'YYYY-MM-DD')
	AND enddate > TO_DATE(:1, 'YYYY-MM-DD')`

`rows, err = r.db.QueryContext(ctx, query, from.Format("2006-01-02"))`

This worked with godror 0.24.2, but fails in 0.30.2
From what version did this stopped working?
So we can upgrade to that version.

**Comment by Mnemonics** [comment]
It also works in 0.26.3

I found that this is actually correct according to Oracle docs.
https://docs.oracle.com/cd/E18283_01/timesten.112/e13070/names.htm

Apparently you can set a connection attribute: DuplicateBindMode

**Reply by Mnemonics** [reply]
It works in 0.29.0, but fails in 0.30.0.
DuplicateBindMode does not work in 0.30.2

**Reply by primananavati** [reply]
Is there any solution on this? I' getting sae issue on 0.36.0 version.

**Reply by tgulacsi** [reply]
DuplicateBindMode is a TimesTen specific parameter.

Either use numeric bind parameters, in strict monotone increasing sequence,
or use parameter names with sql.Named.

**Comment by tgulacsi** [comment]
:1 is for BIND BY POSITION. For bind by name, use a NAME:
```
SELECT column FROM table 
  WHERE  startdate < TO_DATE(:adate, 'YYYY-MM-DD') AND enddate > TO_DATE(:adate, 'YYYY-MM-DD')

rows, err = r.db.QueryContext(ctx, query, sql.Named("adate", from.Format("2006-01-02")))
```

(BTW. Why not
```
SELECT column FROM table 
  WHERE  :1 BETWEEN startdate+1/24/3600 AND enddate-1/24/3600

rows, err = r.db.QueryContext(ctx, query, from)
```
?)
